### DISCUSSION TITLE
How to make my validation data the right size  (SwinUNETR implementation)

### DISCUSSION THREAD

**Discussion by Sshinobu21** [Project-MONAI/tutorials#1699] [Q&A]
Hello,

I'm trying to implement a SwinUNETR model on custom data.
The whole training process seems to be doing great but when I get to the validation test I get an error due to the difference of size between my validation images and their corresponding labels.
I tried to use  RandCropByPosNegLabeld, but I then run into errors to vizualise the data.
Then I tried Resized but it makes my session crash (I'm using google colab)
Does anyone know how to fix this ?

**Comment by KumoLiu** [comment]
Hi @Sshinobu21, hope you can get some guidance in these tutorials.
https://github.com/Project-MONAI/tutorials/blob/main/modules/3d_image_transforms.ipynb
https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/swin_unetr_brats21_segmentation_3d.ipynb

Thanks.

**Reply by Sshinobu21** [reply]
Hi @KumoLiu, thank for your answer but I wasn't able to find a solution
My error happens when i use the dice_metric function. I get the following error: The size of tensor a (407) must match the size of tensor b (1536) at non-singleton dimension 1

**Reply by Sshinobu21** [reply]
Update: my error seems to be resolved by removing this transform from my val_transforms : Spacingd(keys=["image", "label"],pixdim=(1.5, 1.5, 2.0),mode=("bilinear", "nearest")).
These are my transforms now:
val_transforms = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        EnsureChannelFirstd(keys=["image", "label"]),
        CropForegroundd(keys=["image", "label"], source_key="image"),
        EnsureTyped(keys=["image", "label"], device=device, track_meta=True),
    ]
)

However, always in the validation function, the sliding_window_inference function takes forever... is it to be expected?
